test/remoting.integration.js

```diff
@@ -240,3 +183,148 @@ describe('remoting - integration', function() {
+      'patchOrCreate(data:object):storeWithReplaceOnPUTtrue PATCH /stores-replacing',
+      'replaceOrCreate(data:object):storeWithReplaceOnPUTtrue POST /stores-replacing/replaceOrCreate',
+      'replaceOrCreate(data:object):storeWithReplaceOnPUTtrue PUT /stores-replacing',
+      'exists(id:any):boolean GET /stores-replacing/:id/exists',
+      'findById(id:any,filter:object):storeWithReplaceOnPUTtrue GET /stores-replacing/:id',
+      'replaceById(id:any,data:object):storeWithReplaceOnPUTtrue POST /stores-replacing/:id/replace',
+      'replaceById(id:any,data:object):storeWithReplaceOnPUTtrue PUT /stores-replacing/:id',
+      'find(filter:object):storeWithReplaceOnPUTtrue GET /stores-replacing',
+      'findOne(filter:object):storeWithReplaceOnPUTtrue GET /stores-replacing/findOne',
+      'updateAll(where:object,data:object):object POST /stores-replacing/update',
+      'deleteById(id:any):object DELETE /stores-replacing/:id',
+      'count(where:object):number GET /stores-replacing/count',
+      'prototype.patchAttributes(data:object):storeWithReplaceOnPUTtrue PATCH /stores-replacing/:id',
+      'createChangeStream(options:object):ReadableStream POST /stores-replacing/change-stream',
+    ];
+    // The list of methods is from docs:
+    // https://docs.strongloop.com/display/public/LB/Exposing+models+over+REST
+    expect(methods).to.include.members(expectedMethods);
+  });
+});
+
+function formatReturns(m) {
+  var returns = m.returns;
+  if (!returns || returns.length === 0) {
+    return '';
+  }
+  var type = returns[0].type;
+  return type ? ':' + type : '';
+}
+
+function formatMethod(m) {
+  var arr = [];
+  var endpoints = m.getEndpoints();
+  for (var i = 0; i < endpoints.length; i++) {
+    arr.push([
+      m.name,
+      '(',
+      m.accepts.map(function(a) {
+        return a.arg + ':' + a.type;
+      }).join(','),
+      ')',
+      formatReturns(m),
+      ' ',
+      endpoints[i].verb,
+      ' ',
+      endpoints[i].fullPath,
+    ].join(''));
+  }
+  return arr;
+}
+
+function findClass(name) {
+  return app.handler('rest').adapter
+    .getClasses()
+    .filter(function(c) {
+      return c.name === name;
+    })[0];
+}
+
+function getFormattedMethodsExcludingRelations(methods) {
+  return result = methods.filter(function(m) {
+    return m.name.indexOf('__') === -1;
+  })
+  .map(function(m) {
+    return formatMethod(m);
+  })
+  .reduce(function(p, c) {
+    return p.concat(c);
+  });
+}
+
+function getFormattedScopeMethods(methods) {
+  return result = methods.filter(function(m) {
+    return m.name.indexOf('__') === 0;
+  })
+  .map(function(m) {
+    return formatMethod(m);
+  })
+  .reduce(function(p, c) {
+    return p.concat(c);
+  });
+}
+
+function getPrototypeMethods(methods) {
+  return result = methods.filter(function(m) {
+    return m.name.indexOf('prototype.__') === 0;
+  })
+  .map(function(m) {
+    return formatMethod(m);
+  })
+  .reduce(function(p, c) {
+    return p.concat(c);
+  });
+}
```

Comment:
Also please note I have changed the logic for `formatMethod`, `getFormattedMethodsExcludingRelations `, `getFormattedScopeMethods`,... based on the changes in [strong-remoting](https://github.com/strongloop/strong-remoting/pull/305)...